src/main/scala/uk/gov/nationalarchives/tdr/keycloak/KeycloakUtils.scala

```diff
@@ -28,8 +37,32 @@ class KeycloakUtils(url: String) {
   def token(token:String): Option[Token] = {
     getAccessToken(token).map(at => Token(at, new BearerAccessToken(token)))
   }
+
+  def serviceAccountToken(clientId: String, clientSecret: String): Future[BearerAccessToken] = {
+    implicit val backend: SttpBackend[Future, Nothing, WebSocketHandler] = AsyncHttpClientFutureBackend()
+
+    val body: Map[String, String] = Map("grant_type" -> "client_credentials")
+
+    val response: Future[Response[Either[ResponseError[Error], AuthResponse]]] = basicRequest
+      .body(body)
+      .auth.basic(clientId, clientSecret)
+      .post(uri"$url/realms/tdr/protocol/openid-connect/token")
+      .response(asJson[AuthResponse])
+      .send()
+
+    val authResponse = response.flatMap { r =>
+      r.body match {
+        case Right(body) => Future.successful(body)
+        case Left(_) => Future.failed(HttpError("An error occurred contacting the auth server"))
```

Comment:
Is there any useful information inside the `Left` that we should include in the error for debugging purposes?

Reply:
There's not loads of information. The message on the exception is null but there is some information in the exception body which could help us track things. I've changed it to return the exception.